### DISCUSSION TITLE
Move frame by frame through project.

### DISCUSSION THREAD

**Discussion by bobby-sills** [ScratchAddons/ScratchAddons#5019] [Ideas]
Moving frame by frame through a project would be _awesome_ for debugging purposes. I think this would be possible if you made use of the pausing and unpausing addon. This would be extremely useful to a lot of people, and I don't think it's impossible/really hard to implement. Please consider this.

**Comment by Samq64** [comment]
You can use the [debugger addon](scratch.mit.edu/scratch-addons-extension/settings#debugger) while the project is paused and use the button on the threads tab. I'm not sure if that's exactly what you're looking for though.

**Reply by CST1229** [reply]
Doesn't that step block-by-block?

**Reply by Samq64** [reply]
Yeah, I guess it does, so that's not what he's asking.
